rust/main/agents/relayer/src/relayer.rs

```diff
@@ -626,16 +647,21 @@ impl Relayer {
     ) -> Instrumented<JoinHandle<()>> {
         let span = info_span!("SerialSubmitter", destination=%destination);
         let destination = destination.clone();
-        tokio::spawn(TaskMonitor::instrument(&task_monitor, async move {
-            // Propagate task panics
-            serial_submitter.spawn().await.unwrap_or_else(|err| {
-                panic!(
-                    "destination submitter panicked for destination {}: {:?}",
-                    destination, err
-                )
-            });
-        }))
-        .instrument(span)
+        let mut name = "submitter::destination::".to_string();
+        name.push_str(&destination.name());
+        tokio::task::Builder::new()
+            .name(&name)
+            .spawn(TaskMonitor::instrument(&task_monitor, async move {
+                // Propagate task panics
+                serial_submitter.spawn().await.unwrap_or_else(|err| {
+                    panic!(
+                        "destination submitter panicked for destination {}: {:?}",
+                        destination, err
+                    )
+                });
+            }))
+            .expect("spawning tokio task from Builder is infalliable")
```

Comment:
why does it return a Result then? I assume there's some condition in which an error could occur, but it's just unlikely?

Reply:
It is a good question on the API design, but the `spawn` method just cannot fail ever in `tokio::task::Builder`:

```
    pub fn spawn<Fut>(self, future: Fut) -> io::Result<JoinHandle<Fut::Output>>
    where
        Fut: Future + Send + 'static,
        Fut::Output: Send + 'static,
    {
        let fut_size = mem::size_of::<Fut>();
        Ok(if fut_size > BOX_FUTURE_THRESHOLD {
            super::spawn::spawn_inner(Box::pin(future), SpawnMeta::new(self.name, fut_size))
        } else {
            super::spawn::spawn_inner(future, SpawnMeta::new(self.name, fut_size))
        })
    }
```